bolt-simple-taproot.md

```diff
@@ -377,348 +352,276 @@ The Context column decodes as follows:
+  from `bip-musig2`.
+- MUST generate a unique nonce to combine with the `musig2_nonce` previously received in the latest
+  of `channel_ready`, `channel_reestablish`, or `revoke_and_ack` using `NonceAgg` from `bip-musig2`.
+- MUST use the generated secret nonce and the calculated aggregate nonce to construct a
+  `musig2` partial signature for the sender's remote commitment using the `Sign` algorithm from `bip-musig2`.
+- MUST include the partial signature and the public counterpart of the generated nonce in
+  the `partial_signature_with_nonce` field.
+- MUST compute each HTLC signature according to
+  [BIP 342](https://github.com/bitcoin/bips/blob/master/bip-0342.mediawiki), as a BIP 340 Schnorr
+  signature (non-partial).
 
-  - Fail the channel if the received schnorr partial signature failed to verify
-    using the `PartialSigVerify` paramterized with the local combined nonce
-    (`verification_musig2_pubnonce` of the verifying party, and the
-    `signing_musig2_pubnonce` of the sending party) algorithm of `musig2`.
+The recipient:
 
-  - Fail the channel is _any_ of the received HTLC signatures does not validate
-    according to the message extensions and sighash algorithm described in BIP
-    342.
+- MUST fail the channel if `signature` is non-zero.
+- MUST fail the channel if `partial_signature_with_nonce` is absent.
+- MUST compute the aggregate nonce from:
+  - the `musig2_nonce` field the recipient previously sent in the latest of `channel_ready`,
+    `channel_reestablish`, or `revoke_and_ack`
+  - the `public_nonce` included as part of the `partial_signature_with_nonce` field
+- MUST verify the `partial_signature_with_nonce` field using the `PartialSigVerifyInternal`
+  algorithm of `bip-musig2`:
+  - if the partial signature is invalid, MUST fail the channel
+- MUST fail the channel if _any_ of the received HTLC signatures is invalid according to BIP 342.
 
 #### `revoke_and_ack` Extensions
 
 A new TLV stream is added to the `revoke_and_ack` message:
 
-1. type: 133 (`revoke_and_ack`)
-2. data:
-    * ...
-    * [`revoke_and_ack_tlvs`:`tlvs`]
-
 1. `tlv_stream`: `revoke_and_ack_tlvs`
 2. types:
-    1. type: 2 (`verification_musig2_pubnonce`)
-    2: data:
-        * [`66*byte`:`nonces`]
-
-Both sides **MUST** provide this new TLV field.
+   1. type: 4 (`musig2_nonce`)
+   2. data:
+      * [`66*byte`: `public_nonce`]
 
-Similar to sending the `next_per_commitment_point`, we also send the _next_ set
-of musig2 nonces after we revoke a state. Sending these nonces allows the
-remote party to propose another state transition as soon as the message is
+Similar to sending the `next_per_commitment_point`, we also send the _next_ `musig2` nonce after we revoke a
+state. Sending this nonce allows the remote party to propose another state transition as soon as the message is
 received.
 
-#### `channel_reestablish` Extensions
```

Comment:
I don't think this section can just be removed all together. 

As is, these changes don't mention how to handle retransmission at at all. Remember that it's possible that a node sends a state during retransmission, but with a distinct set of HTLCs (one of them timed out or w/e). If that _same_ nonce is used to send this extra sig, then the other party can lie that they didn't get the sig and force you to re-use the nonce.

Reply:
We split it back out from `channel_ready`.